webserver/jersey/src/main/java/io/helidon/webserver/jersey/ResponseWriter.java

```diff
@@ -327,5 +328,30 @@ private void awaitRequest() throws IOException {
                 throw new IOException("Bad news: the stream has been closed");
             }
         }
+
+        // We should allow to override these values via config
+        private static final int MIN_WAIT = 100;
+        private static final int MAX_WAIT = 1600;
+        private static final long MEMORY_LIMIT_MB = 256 * 1024 * 1024;
+
+        private void ensureBoundedMemory() {
+            int millis = MIN_WAIT;
+            while (millis <= MAX_WAIT
+                    && (PooledByteBufAllocator.DEFAULT.pinnedHeapMemory() > MEMORY_LIMIT_MB
+                            || PooledByteBufAllocator.DEFAULT.pinnedDirectMemory() > MEMORY_LIMIT_MB)) {
+                try {
+                    LOGGER.finest("Waiting " + millis + " to allocate another ByteBuf");
+                    Thread.sleep(millis);
+                    millis *= 2;        // exponential backoff
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
```

Comment:
(`Thread.currentThread().interrupt()` before rethrowing or the interrupt status will get lost)